src/main/java/seedu/address/model/game/GameMode.java

```diff
@@ -0,0 +1,75 @@
+package seedu.address.model.game;
+
+import seedu.address.model.task.Status;
+import seedu.address.model.task.Task;
+
+// @@author chikchengyao
+
+/**
+ * A GameMode is a procedure for determining the exact XP amount gained when completing tasks.
+ * Each particular method extends GameMode to describe its particular method of calculation.
+ */
+public abstract class GameMode {
+
+    /**
+     * Checks that the two supplied tasks are the same task, i.e., they are identical except
+     * for completion status.
+     *
+     * @param task1 First task to be compared.
+     * @param task2 Second task to be compared.
+     */
+    void checkTaskDetails(Task task1, Task task2) throws XpEvaluationException {
+        if (!task1.getName().fullName.equals(task2.getName().fullName)) {
+            throw new XpEvaluationException("Tasks have different name.");
+        }
+
+        if (!task1.getDueDate().valueDate.equals(task2.getDueDate().valueDate)) {
+            throw new XpEvaluationException("Tasks have different due date.");
+        }
+
+        if (!task1.getPriorityValue().value.equals(task2.getPriorityValue().value)) {
+            throw new XpEvaluationException("Tasks have different priority value.");
+        }
+
+        if (!task1.getDescription().value.equals(task2.getDescription().value)) {
```

Comment:
if i'm not wrong, the unique identifier of a task is its name, due date and priority value, so it's impossible for two tasks to have all of the above the same but different description, this check might not be necessary?